src/components/backgroundselector/BackgroundSelectorDirective.js

```diff
@@ -95,11 +97,24 @@ goog.require('ga_topic_service');
             dereg();
           });
 
-          scope.$on('gaTopicChange', function(event, newTopic) {
-            // If the layers config is loaded
-            if (gaLayers.getBackgroundLayers()) {
-              updateBgLayer(newTopic);
-            }
+          gaTopic.getTopics().then(function() {
+            scope.$on('gaLayersChange', function() {
```

Comment:
We don't have garantueed order of exection between the getTopics promise and the gaLayersChange event; so I guess that this implementation might break when topics take longer to load than the layers configuration. Also, gaLayersChange will be called on every language change, not only once at startup. But it won't get called on topic change!

Not sure how to address this. I need to understand more. What's the difference between the gaLayersChange part here and the one above?